templates/remix/app/entry.server.tsx

```diff
@@ -14,6 +14,17 @@ export default function handleRequest(
 
   responseHeaders.set("Content-Type", "text/html");
 
+  // Add Link header for HTTP/2 Server Push
+  let http2PushLinksHeaders = remixContext.matches
+    .flatMap(({ route: { module, imports } }) => [module, ...(imports || [])])
+    .filter(Boolean)
+    .concat([
+      remixContext.manifest.url,
+      remixContext.manifest.entry.module,
+      ...remixContext.manifest.entry.imports,
+    ]);
+  responseHeaders.set("Link", (responseHeaders.has("Link") ? responseHeaders.get("Link") + "," : '') + http2PushLinksHeaders.map((link: string) => `<${link}>; rel=preload; as=script`).join(","));
```

Comment:
This could be simplified to

```suggestion
  responseHeaders.set(
    "Link",
    http2PushLinksHeaders
      .map((link: string) => `<${link}>; rel=preload; as=script`)
      .concat(responseHeaders.get("Link") as string)
      .filter(Boolean)
      .join(",")
  );
```

Reply:
less fan of that change, it's less readable to me, but I changed it ;) Thanks @MichaelDeBoey

Reply:
This is more clear than all the `+` signs imo.

- You're mapping the created array + adding the already existing `Link` header
- You're removing the possible `null` value (if the `Link` header didn't exist)
- You're joining the whole array